docs/sequence.rst

```diff
@@ -0,0 +1,647 @@
+    -- = fromList [1*10, 2*10, 3*10]
+    > fromList [10,20,30]
+
+    (*10) <$> Seq.fromList [1, 2, 3]
+    -- = fromList [1*10, 2*10, 3*10]
+    > fromList [10,20,30]
+
+    let myMapFunc index val = index * val
+
+    Seq.mapWithIndex myMapFunc (Seq.fromList [1, 2, 3])
+    -- = fromList [0*1, 1*2, 2*3]
+    > fromList [0,2,6]
+
+
+Delete an element
+"""""""""""""""""
+
+::
+
+    Seq.deleteAt :: Int -> Seq a -> Seq a
+    Seq.deleteAt i xs = ...
+
+:seq:`deleteAt` removes the element of the sequence at index ``i``. If the index
+is out of bounds then the original sequence is returned.
+
+::
+
+    Seq.deleteAt 0 (Seq.fromList [0, 1, 2])
+    > fromList [1,2]
+
+    Seq.deleteAt 10 (Seq.fromList [0, 1, 2])
+    > fromList [0,1,2]
+
+Sorting
+^^^^^^^
+
+::
+
+    Seq.sort :: Ord a => Seq a -> Seq a
+    Seq.sort xs = ...
+
+:seq:`sort` the sequence ``xs`` using the ``Ord`` instance.
+
+::
+
+    Seq.sort (Seq.fromList ["x", "a", "c", "b"])
+    > fromList ["a","b","c","x"]
+
+
+Subsequences
+^^^^^^^^^^^^
+
+Take
+""""
+
+::
+
+    Seq.take :: Int -> Seq a -> Seq a
+    Seq.take n xs = ...
+
+:seq:`take` returns the first ``n`` elements of the sequence ``xs``. If the
+length of ``xs`` is less than ``n`` then all elements are returned.
+
+::
+
+    Seq.take 0 (Seq.fromList [1, 2, 3])
+    > fromList []
+
+    Seq.take 2 (Seq.fromList [1, 2, 3])
+    > fromList [1,2]
+
+    Seq.take 5 (Seq.fromList [1, 2, 3])
+    > fromList [1,2,3]
+
+Drop
+""""
+
+::
+
+    Seq.drop :: Int -> Seq a -> Seq a
+    Seq.drop n xs = ...
+
+:seq:`drop` the first ``n`` elements of the sequence ``xs``. If the length of
+``xs`` is less than ``n`` then an empty sequence is returned.
+
+::
+
+    Seq.drop 0 (Seq.fromList [1, 2, 3])
+    > fromList [1,2,3]
+
+    Seq.drop 2 (Seq.fromList [1, 2, 3])
+    > fromList [3]
+
+    Seq.drop 5 (Seq.fromList [1, 2, 3])
+    > fromList []
+
+Chunks
+""""""
+
+::
+
+    Seq.chunksOf :: Int -> Seq a -> Seq (Seq a)
+    Seq.chunksOf k xs = ...
+
+:seq:`chunksOf` splits the sequence ``xs`` into chunks of size ``k``. If the
+length of the sequence is not evenly divisible by ``k`` then the last chunk will
+have less than ``k`` elements.
+
+.. WARNING::
+   ``k`` can only be ``0`` when the sequence is empty, otherwise a runtime
+   ``error`` is thrown.
+
+::
+
+    -- A chunk size of 0 can ONLY be given for an empty sequence.
+    Seq.chunksOf 0 Seq.empty
+    > fromList []
+
+    Seq.chunksOf 1 (Seq.fromList [1, 2, 3])
+    > fromList [fromList [1],fromList [2],fromList [3]]
+
+    Seq.chunksOf 2 (Seq.fromList [1, 2, 3])
+    > fromList [fromList [1,2],fromList [3]]
+
+    Seq.chunksOf 5 (Seq.fromList [1, 2, 3])
+    > fromList [fromList [1,2,3]]
+
+
+Folding
+^^^^^^^
+
+::
+
+    foldr :: (a -> b -> b) -> b -> Seq a -> b
+    foldr f init xs = ...
+
+    Seq.foldrWithIndex :: (Int -> a -> b -> b) -> b -> Seq a -> b
+    Seq.foldrWithIndex f init xs = ...
+
+:seq:`foldr` collapses the sequence into a summary value by applying repeatedly
```

Comment:
"applying repeatedly applying". Pick one and only one applying.

Reply:
:)